client/src/lib.rs

```diff
@@ -1,146 +1,63 @@
-    async fn get_treasury_all_fungible_token_balance(
+use async_trait::async_trait;
+use execution::*;
+use eyre::Error;
+use merkle_tree::MerkleProof;
+use rust_decimal::Decimal;
+use simperby_common::*;
+use simperby_settlement::*;
+
+pub struct Ethereum {}
+
+#[async_trait]
+impl SettlementChain for Ethereum {
+    async fn get_chain_name(&self) -> String {
+        todo!()
+    }
+
+    async fn check_connection(&self) -> Result<(), Error> {
+        todo!()
+    }
+
+    async fn get_last_block(&self) -> Result<SettlementChainBlock, Error> {
+        todo!()
+    }
+
+    async fn get_relayer_account_info(&self) -> Result<(String, Decimal), Error> {
+        todo!()
+    }
+
+    async fn get_light_client_header(&self) -> Result<BlockHeader, Error> {
+        todo!()
+    }
+
+    async fn get_treasury_fungible_token_balance(
         &self,
-    ) -> Result<HashMap<String, Decimal>, Error>;
+        _address: String,
+    ) -> Result<Decimal, Error> {
+        todo!()
+    }
 
-    /// Returns the current balance of a particular non-fungible token collection in the treasury contract,
-    /// identified as their token indices.
     async fn get_treasury_non_fungible_token_balance(
         &self,
-        address: Vec<String>,
-    ) -> Result<String, Error>;
+        _address: String,
+    ) -> Result<Vec<String>, Error> {
+        todo!()
+    }
 
-    /// Returns the current balance of all non-fungible tokens in the treasury contract,
-    /// identified as `(collection address, token index)`.
-    async fn get_treasury_all_non_fungible_token_balance(
-        &self,
-    ) -> Result<Vec<(String, String)>, Error>;
-
-    /// Updates the light client state in the treasury by providing the next, valid block header and its proof.
-    ///
-    /// This is one of the message delivery methods; a transaction that carries the given data will be submitted to the chain.
     async fn update_treasury_light_client(
         &self,
-        header: Header,
-        proof: BlockFinalizationProof,
-    ) -> Result<(), Error>;
-
-    /// Transfers a given amount of fungible tokens from the treasury contract to the destination address.
-    ///
-    /// This is one of the message delivery methods; a transaction that carries the given data and the proof will be submitted to the chain.
-    async fn transfer_treasury_fungible_token(
-        &self,
-        message: FungibleTokenTransfer,
-        block_height: u64,
-        proof: MerkleProof,
-    ) -> Result<(), Error>;
+        _header: BlockHeader,
+        _proof: FinalizationProof,
+    ) -> Result<(), Error> {
+        todo!()
+    }
 
-    /// Transfers an NFT from the treasury contract to the destination address.
-    ///
-    /// This is one of the message methods; a transaction that carries the given data and the proof will be submitted to the chain.
-    async fn transfer_treasury_non_fungible_token(
+    async fn execute(
```

Comment:
Do the functions related to treasury now run through this?

Reply:
Yes.